apps/bluetooth-music/app.js

```diff
@@ -0,0 +1,98 @@
+'use strict'
+
+var logger = require('logger')('bluetooth-music')
+var Application = require('@yodaos/application').Application
+var bluetooth = require('@yoda/bluetooth')
+var protocol = bluetooth.protocol
+var AudioFocus = require('@yodaos/application').AudioFocus
+var audioFocus = null
+var a2dp = null
+
+function onAudioFocusGained () {
+  logger.debug('onAudioFocusGained')
+  a2dp.unmute()
+}
+
+function onAudioFocusLost () {
+  logger.debug('onAudioFocusLost')
+  a2dp.mute()
+  app.finalize()
+}
+
+function onAudioStateChangedListener (mode, state, extra) {
+  logger.debug(`${mode} onAudioStateChanged(${state})`)
+  switch (state) {
+    case protocol.AUDIO_STATE.PLAYING:
+      break
+    case protocol.AUDIO_STATE.PAUSED:
+    case protocol.AUDIO_STATE.STOPPED:
+      audioFocus.abandon()
+      break
+    case protocol.AUDIO_STATE.VOLUMN_CHANGED:
+      break
+  }
+}
+
+var app = Application({
+  created: () => {
+    logger.debug('created')
+    audioFocus = new AudioFocus(AudioFocus.TRANSIENT_EXCLUSIVE)
+    audioFocus.onGain = onAudioFocusGained
+    audioFocus.onLoss = onAudioFocusLost
+    audioFocus.request()
+    a2dp = bluetooth.getAdapter(protocol.PROFILE.A2DP)
+    a2dp.on('audio_state_changed', onAudioStateChangedListener)
+  },
+  destroyed: () => {
+    logger.debug('destroyed')
+    a2dp.destroy()
+  },
+  url: (url) => {
+    logger.debug('on url: ', url)
+    if (url.hostname !== 'bluetooth-music') {
+      return
+    }
+    var intent = url.pathname.substr(1)
+    logger.debug(`intent = ${intent}`)
+    switch (intent) {
+      case 'playing':
+        break
+      case 'next':
+        a2dp.next()
+        break
+      case 'pre':
+        a2dp.prev()
+        break
+      case 'stop':
+        a2dp.stop()
+        break
+      case 'resume':
+        a2dp.play()
+        break
+      case 'like':
+        break
+      case 'bluetooth_info':
+        break
+      default:
+        break
+    }
+  },
+  broadcast: action => {
```

Comment:
Suggest to keep consistent with the broadcast API, use `channel`.

Reply:
keep original broadcast/channel naming style.